client/my-sites/themes/install-theme-button.jsx

```diff
@@ -6,79 +6,82 @@ import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
-import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
-import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import {
+	isJetpackSiteMultiSite,
+	isJetpackSite,
+	getSiteAdminUrl,
+	getSiteSlug,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
 
 import './install-theme-button.scss';
 
+function getInstallThemeUrl( state, siteId ) {
+	if ( ! siteId ) {
+		return '/themes/upload';
+	}
+
+	const atomicSite = isAtomicSite( state, siteId );
+	const siteCanInstallThemes = siteHasFeature( state, siteId, FEATURE_INSTALL_THEMES );
+	if ( atomicSite && siteCanInstallThemes ) {
+		return getSiteAdminUrl( state, siteId, 'theme-install.php' );
+	}
+
+	const siteSlug = getSiteSlug( state, siteId );
+	return `/themes/upload/${ siteSlug }`;
+}
+
+function getSiteType( atomicSite, jetpackSite, selectedSiteId ) {
+	if ( atomicSite ) {
+		return 'atomic';
+	} else if ( jetpackSite ) {
+		return 'jetpack';
+	} else if ( selectedSiteId ) {
+		return 'simple';
+	}
+
+	return null;
+}
+
 const InstallThemeButton = ( {
+	isLoggedIn,
+	selectedSiteId,
 	isMultisite,
+	atomicSite,
 	jetpackSite,
-	isLoggedIn,
-	siteSlug,
-	siteCanInstallThemes,
+	installThemeUrl,
 	dispatchTracksEvent,
-	atomicSite,
 } ) => {
 	if ( ! isLoggedIn || isMultisite ) {
 		return null;
 	}
 
-	let siteType = null;
-	if ( ! isLoggedIn ) {
-		siteType = 'logged_out';
```

Comment:
While using `logged_out` for a `siteType` variable value doesn't make much sense from a dev perspective, it was used in an existing tracks event, so there is a possibility that we might break some funnels or dashboards. Should we keep it instead?

Reply:
When `isLoggedIn` is `false` (or when on a multisite install), the component function will return `null` a few lines above -- no install button is rendered at all. Further `isLoggedIn` logic can be removed.